### DISCUSSION TITLE
Glossary revision workflow

### DISCUSSION THREAD

**Discussion by cmutel** [Depart-de-Sentier/sentier.dev#3] [General]
The glossary structure can be thought of as a managed forest. The soil is our common set of labels for products, industries, units, and locations. Each tree is a specific industry sector, and has its own set of terms namespaced to that specific industry. Some trees are bigger than others; our job as foresters is to manage their size so that they don't become too complex to use, and to that they don't interfere with other healthy trees. Models can refer to industry-specific terms, and the general terms (the tree's root system).

We will start with predefined terms, including:

* Common nomenclature (CN - products)
* Harmonized system (HS - products)
* Central product classification (CPC - products)
* International Standard Industrial Classification of All Economic Activities (ISIC - industries)
* Statistical Classification of Economic Activities in the European Community (NACE - industries)
* Geonames (places)
* Natural Earth (places)
* Quantities, Units, Dimensions and Types (QUDT - units)
* Ontology of units of Measure and related concepts (OM - units)

As time develops, and especially as the first models are integrated, the industry-specific trees will need to grow.

Open questions:

- What kind of permissions and workflows are needed for introducing and merging revisions?
- What does the release process look like, and how often is there a release?
- What is our long-term vision on automatic trust in the system?
- How are users notified of additions and changes?

**Comment by mklarmann** [comment]
Eaternity currently uses GADM for locations (like Hestia). And we have food certifications, physical units and foodex2 as terms. We are still looking for a harmonized set of terms for processes. Our hope was to be able to extract this from ecoinvent? or Hestia?

**Reply by cmutel** [reply]
I would be careful with GADM, their license is not your friend: https://gadm.org/license.html

> FoodEx2 consists of descriptions of a large number of individual food items aggregated into food groups and broader food categories in a hierarchical parent-child relationship.

From the [EFSA](https://www.efsa.europa.eu/en/data/data-standardisation). A good addition, thanks.

>  food certifications, physical units

Are these specific systems or generic concepts? I can't search and find anything helpful here 😛

**Reply by mklarmann** [reply]
We did not use a generic concept for physical units yet (other then checking wikipedia). But I think in some conversation you mentioned: https://github.com/ucum-org/ucum 

Regarding GADM, we also considered using [google](https://developers.google.com/maps/documentation/geocoding/requests-geocoding?hl=de) and/or https://earthengine.google.com , yet their dataset did not some strictly hierarchical. And we considered aligning with Hestia on this one.

**Comment by jsvgoncalves** [comment]
- Do we create our own classification (very strongly not), i.e. we merge the tree early, or do we just map in between classifications?
- HS and CPC have *-* relationships (in all combinations); also the granularity level is very different, do we have a plan to deal with this?



Do use heroes system for specific industries/classification systems?

Do we follow the stackoverflow model reputation model somehow? 

Should everyone be able to propose changes? Maybe at least by default; we should probably find "heroes" for each industry and give them higher permission levels
Do we want to follow a git/github approach, where you push changes to fork/branch, and you don't necessarily need to merge them to `main` to pull those changes locally?

Should we provide notifications not only for changes, but also proposed changes (PRs)
Should our data merging process follow a more RFC-like approach allowing enough time for comments from interested parties?
Should we automatically notify people if a data change affects their model, the terms their model is using etc?
We need to understand _what_ changed, is it on a classification level or on a specific sector?
If you contribute, do you have access to certain features?
We should make the first contribution smooth and rewarding (either by reputation or by features within the business model)
Do we make copies of all URI entries from other systems? Do we create our own on import and map it to the existing one?
